src/core/layout/qgslayoutrendercontext.h

```diff
@@ -297,6 +297,18 @@ class CORE_EXPORT QgsLayoutRenderContext : public QObject
      */
     QVector<qreal> predefinedScales() const { return mPredefinedScales; }
 
+    /**
+     * Returns the possibly NULL featureFilterProvider
+     * \since QGIS 3.18
+     */
+    QgsFeatureFilterProvider *featureFilterProvider() const;
+
+    /**
+     * Sets \a featureFilterProvider
```

Comment:
These are very "qgis 2.x" style docs 🙃 Can you add more details here about exactly what this is used for and when it's set?